pkg/base/helm.go

```diff
@@ -271,3 +274,105 @@ func checkChartForVersion(file *upstreamtypes.UpstreamFile) (string, error) {
 	// if no determination is made, assume v2
 	return "v2", nil
 }
+
+// insert namespace if it's defined in the spec and not already present in the manifests
+func kustomizeHelmNamespace(baseFiles []BaseFile, renderOptions *RenderOptions) ([]BaseFile, error) {
+	var updatedBaseFiles []BaseFile
+	if renderOptions.Namespace != "" {
+		chartsPath, err := ioutil.TempDir("", "charts")
+		if err != nil {
+			return nil, errors.Wrap(err, "failed to create temp dir")
+		}
+		defer os.RemoveAll(chartsPath)
+
+		resources := map[string]BaseFile{}
+		var kustomizeResources []string
+		var kustomizePatches []kustomizetypes.PatchStrategicMerge
+		foundGVKNamesMap := map[string]bool{}
+		for _, baseFile := range baseFiles {
+			// write temp files for manifests that need a namespace
+			gvk, manifest := GetGVKWithNameAndNs(baseFile.Content, renderOptions.Namespace)
+			if manifest.APIVersion == "" && manifest.Kind == "" && manifest.Metadata.Name == "" {
```

Comment:
since this is unmarshalling into a struct now, it has to check for the common k8s resource fields to identify invalid manifests. this also means that "valid" yaml will be filtered out but that's probably okay considering they'll fail in deployment anyway. let me know if this is a problem though.